### PR TITLE
fix delete platform and add onUnregisteredPlatform method

### PR DESCRIPTION
**Pull request by Rafa97485**
Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/ltijs/lib/Provider/Provider.d.ts
Comment (paulschwoerer) [inline types/ltijs/lib/Provider/Provider.d.ts:45]: I'd use a better fitting name here. In the original library, `unregisteredPlatformCallback` is used. What do you think?

File: types/ltijs/lib/Provider/Provider.d.ts
Comment (paulschwoerer) [inline types/ltijs/lib/Provider/Provider.d.ts:77]: Same as above: I'd use a better fitting name here. In the original library, unregisteredPlatformCallback is used. What do you think?

**Review by paulschwoerer — COMMENTED** [review]
Added comments for you to review. Let me know what you think :)

File: types/ltijs/lib/Provider/Provider.d.ts
Comment (Rafa97485) [reply types/ltijs/lib/Provider/Provider.d.ts:45]: Perfect I will change it

File: types/ltijs/lib/Provider/Provider.d.ts
Comment (Rafa97485) [reply types/ltijs/lib/Provider/Provider.d.ts]: I changed it

**Review by paulschwoerer — APPROVED** [review]
LGTM

**Comment by Rafa97485** [comment]
@paulschwoerer when are you going to merge this pull request and when are we going to have these changes in the public library?

**Comment by paulschwoerer** [comment]
Ah I see, I cannot merge this change. This is because there are no tests for your changes. The typescript bot states above that only maintainers can merge such changes. 
Is it possible for you to add tests?

**Comment by peterblazejewicz** [comment]
please add a test, as per readme

**Comment by Rafa97485** [comment]
I tried to run tests for ltijs and I had this error
D:\types-ltijs\LtijsTypes> npm test ltijs

> ***@***.*** test
> node --require source-map-support/register
***@***.***/dtslint/ types "ltijs"

***@***.***
Error: D:\ItopiaRafa\types-ltijs\LtijsTypes\types\ltijs: Missing
'package.json'
    at checkPackageJson
***@***.***\dtslint\src\checks.ts:15:11)
    at runTests
***@***.***\dtslint\src\index.ts:146:39)
    at main
***@***.***\dtslint\src\index.ts:88:5)

**Comment by Rafa97485** [comment]
I tried to run tests for ltijs and I had this error
D:\types-ltijs\LtijsTypes> npm test ltijs

> ***@***.*** test
> node --require source-map-support/register
***@***.***/dtslint/ types "ltijs"

***@***.***
Error: D:\ItopiaRafa\types-ltijs\LtijsTypes\types\ltijs: Missing
'package.json'
    at checkPackageJson
***@***.***\dtslint\src\checks.ts:15:11)
    at runTests
***@***.***\dtslint\src\index.ts:146:39)
    at main
***@***.***\dtslint\src\index.ts:88:5)

**Comment by Rafa97485** [comment]
I had this error with the first command ERROR --workspace-root may only be
used inside a workspace

**Comment by Rafa97485** [comment]
I am still having the same error
PS D:\ItopiaRafa\types-ltijs\LtijsTypes> npx pnpm test ltijs

> ***@***.*** test D:\ItopiaRafa\types-ltijs\LtijsTypes
> node --require source-map-support/register
***@***.***/dtslint/ types "ltijs"

***@***.***
Error: D:\ItopiaRafa\types-ltijs\LtijsTypes\types\ltijs: Missing
'package.json'
    at checkPackageJson
***@***.***\dtslint\src\checks.ts:15:11)
    at runTests
***@***.***\dtslint\src\index.ts:149:39)
    at main
***@***.***\dtslint\src\index.ts:88:5)
ELIFECYCLE Test failed. See above for more details.

**Comment by Rafa97485** [comment]
I can't fix the whole package. I really need to merge this PR. The unit
tests don't work this is the error:  Error:
D:\ItopiaRafa\types-ltijs\LtijsTypes\types\ltijs:
Missing 'package.json'.

**Comment by Rafa97485** [comment]
I can't fix the whole package. I really need to merge this PR. The unit
tests don't work this is the error:  Error:
D:\ItopiaRafa\types-ltijs\LtijsTypes\types\ltijs:
Missing 'package.json'.

**Review by peterblazejewicz — APPROVED** [review]

**Comment by Rafa97485** [comment]
Ready to merge

**Comment by paulschwoerer** [comment]
Bad luck with those package errors :(
 Thanks for the contribution @Rafa97485

**Comment by Rafa97485** [comment]
You’re welcome

**Pull request merged**
